tee-worker/omni-executor/parentchain/listener/src/event_handler.rs

```diff
@@ -201,26 +122,6 @@ impl<
 			.map(|f| scale_decode::Field::new(f.ty.id, f.name.as_deref()));
 
 		match variant.name.as_str() {
-			IntentRequested::EVENT => {
-				let intent_requested: IntentRequested = IntentRequested::decode_as_fields(
-					&mut event.field_bytes.as_slice(),
-					&mut fields.clone(),
-					metadata.types(),
-				)
-				.map_err(|_| {
-					log::error!("Could not decode event {:?}", event.id);
-					Error::NonRecoverableError
-				})?;
-
-				handle_intent_requested_event(
-					&self.ethereum_intent_executor,
-					&self.solana_intent_executor,
-					&self.rpc_client_factory,
-					self.transaction_signer.clone(),
-					intent_requested,
-				)
-				.await?;
-			},
 			AccountStoreUpdated::EVENT => {
```

Comment:
It handles only this variant `AccountStoreUpdated::EVENT` while it logs `log::debug!("Got IntentRequested event: {:?}", event.id);`. 

Which component is going to handle `IntentRequestd` events ?

Reply:
>  It handles only this variant AccountStoreUpdated::EVENT while it logs log::debug!("Got IntentRequested event: {:?}", event.id);.

I didn't see that log, it's removed now.

> Which component is going to handle IntentRequestd events ?

We are executing the intents directly when handling `NativeCall::request_intent`. We still send the extrinsic to the chain to log the event, but since the omni-executor handles the intents now, there is no need to wait for those events.

Reply:
It's [here](https://github.com/litentry/heima/blob/1bd93e6b0a70eb314ea8927c70debeb7a36f48d6/tee-worker/omni-executor/parentchain/listener/src/event_handler.rs#L98)

Reply:
> We are executing the intents directly when handling NativeCall::request_intent. We still send the extrinsic to the chain to log the event, but since the omni-executor handles the intents now, there is no need to wait for those events.

But from pallet user point of view he can request intent by calling pallet directly, should this call be restricted to enclaves only ?

Reply:
>It's [here](https://github.com/litentry/heima/blob/1bd93e6b0a70eb314ea8927c70debeb7a36f48d6/tee-worker/omni-executor/parentchain/listener/src/event_handler.rs#L98)

What I meant was that I removed it. (https://github.com/litentry/heima/pull/3271/commits/7ead1f0ae4aed8849e8b8b05ca304cdbc224b38e)

>But from pallet user point of view he can request intent by calling pallet directly, should this call be restricted to enclaves only ?

It is already restricted. The `request_intent` call  only supports `OmniAccountOrigin`, so it can only be called from `dispatch_as_omni_account` (which only supports `TEECallOrigin`)

Reply:
Now that I think about it, I don't think we even need the `request_intent` call anymore, we are already logging every call dispatched for omni accounts and as well as emitting events when intents are executed